gateway/api/v1/views/get_provider_jobs.py

```diff
@@ -0,0 +1,167 @@
+    return create_paginated_response(
+        data=serializer.data,
+        total_count=total_count,
+        request=request,
+        limit=limit,
+        offset=offset,
+    )
+
+
+@swagger_auto_schema(
+    method="get",
+    operation_description="List provider Jobs",
+    responses={
+        status.HTTP_200_OK: v1_serializers.JobSerializerWithoutResult(many=True),
+        status.HTTP_400_BAD_REQUEST: openapi.Response(
+            description="Bad Request - Invalid parameters",
+            schema=openapi.Schema(
+                type=openapi.TYPE_OBJECT,
+                properties={
+                    "message": openapi.Schema(
+                        type=openapi.TYPE_STRING, description="Error message"
+                    )
+                },
+            ),
+        ),
+    },
+    manual_parameters=[
+        openapi.Parameter(
+            "limit",
+            openapi.IN_QUERY,
+            description="Number of results to return per page (max: 1000, default: from settings)",
+            type=openapi.TYPE_INTEGER,
+            minimum=0,
+            default=settings.REST_FRAMEWORK["PAGE_SIZE"],
+        ),
+        openapi.Parameter(
+            "offset",
+            openapi.IN_QUERY,
+            description="Number of results to skip before starting to collect results",
+            type=openapi.TYPE_INTEGER,  # pylint: disable=duplicate-code
+            minimum=0,
+        ),
+        openapi.Parameter(
+            "provider",
+            openapi.IN_QUERY,
+            description="Function Provider name",
+            type=openapi.TYPE_STRING,
+        ),
+        openapi.Parameter(
+            "function",
+            openapi.IN_QUERY,
+            description="Function title",
+            type=openapi.TYPE_STRING,
+        ),
+    ],
+)
+@endpoint("jobs/provider")
+@api_view(["GET"])
+@permission_classes([permissions.IsAuthenticated])
+def get_provider_jobs(request):
+    """
+    Return a list of jobs for a given provider/function.
+
+    Query parameters:
+    - limit: results per page
+    - offset: results to skip
+    - provider: provider name (required)
+    - function: function title (required)
+    """
+    try:
+        input_data = serialize_input(request)
+    except ValueError as e:
+        return Response({"message": str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+    try:
+        jobs, total = GetProviderJobsUseCase(**input_data).execute()
+    except ProviderNotFoundException:
+        return Response(
+            {"message": f"Provider {input_data['provider']} doesn't exist."},
+            status=status.HTTP_404_NOT_FOUND,
+        )
+    except FunctionNotFoundException:
+        return Response(
+            {
+                "message": f"Qiskit Function {input_data['provider']}/{input_data['function_name']} doesn't exist."  # pylint: disable=line-too-long
+            },
+            status=status.HTTP_404_NOT_FOUND,
+        )
```

Comment:
In the files migration i created a generic `NotFoundException` in the domain folder: https://github.com/Qiskit/qiskit-serverless/pull/1696/files#diff-e6da6866795b4c157d9b9e887eb27622f036a9dc512d30e5097a301df1d5001c

Also I created an additional decorator to quickly manage the possible `NotFoundException`s (and other that can be created later) and create the corresponding `Response`. The only cons is that you have to set the message when you create the exception to inform about the issue.